homeassistant/components/sensor/kwb.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for KWB Easyfire.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.kwb/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_PORT)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pykwb==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 23
+DEFAULT_TYPE = 'tcp'
+DEFAULT_RAW = 'off'
+
+MODE_SERIAL = 0
+MODE_TCP = 1
+
+CONF_TYPE = 'type'
+CONF_RAW = 'raw'
+
+SERIAL_SCHEMA = {
+    vol.Required(CONF_PORT): cv.string,
```

Comment:
Instead of `CONF_PORT` for serial mode, can you use `CONF_DEVICE` from homeassistant.conf? That should be a little less confusing that reusing the same config field, and then you won't need to specify the type, since you can check if either `host` or `device` is specified.